.github/actions/verify-badge/action.yaml

```diff
@@ -0,0 +1,257 @@
+
+        POD_SIDECAR_COUNT=$(kubectl get pods -n "${NAMESPACE}" -o json | jq '.items[].spec.containers[] | select(.name=="istio-proxy") | length' | wc -l)
+
+        if [ "$POD_COUNT" -ne "$POD_SIDECAR_COUNT" ]; then
+          echo "❌ Not all pods have the istio sidecar"
+          exit 1
+        fi
+        echo "✅ All pods have the istio sidecar"
+
+        echo
+
+        ## Should avoid workarounds such as disabling strict mTLS peer authentication.
+
+        PEERAUTH=$(kubectl get Peerauthentication -n "${NAMESPACE}" -o=json | jq -r '.items[].spec.mtls.mode')
+
+        if [ "$PEERAUTH" != "STRICT" ] && [ "$PEERAUTH" != "" ]; then
+          echo "❌ Peerauth is not strict"
+          exit 1
+        fi
+        echo "✅ Peerauthentication is set to strict"
+
+        echo
+
+        # Network Policies
+
+        ## Must define network policies under the allow key as required.
+
+        NETPOL_AMOUNT=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.network.allow | length')
+
+        if [ "$NETPOL_AMOUNT" -eq 0 ]; then
+          echo "❌ No network policies defined"
+          exit 1
+        fi
+        echo "✅ Network policies are defined"
+
+        ## Should minimize network policies to specific selectors needed for Ingress/Egress traffic.
+
+        NETPOL_EXT_COUNT=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.network.allow[] | select(.remoteGenerated != "IntraNamespace") | length' | wc -l)
+        echo "ℹ️ Non-IntraNamespace network policies: $NETPOL_EXT_COUNT"
+
+        NETPOL_EXT_SELECTOR_COUNT=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.network.allow[] | select(.remoteGenerated != "IntraNamespace") | select(.selector != null) | length' | wc -l)
+        echo "ℹ️ Non-IntraNamespace network policies with selectors: $NETPOL_EXT_SELECTOR_COUNT"
+
+        if [ "$NETPOL_EXT_COUNT" -ne "$NETPOL_EXT_SELECTOR_COUNT" ]; then
+          echo "❌ Not all applicable network policies are using selectors"
+          exit 1
+        fi
+        echo "✅ All applicable network policies are using selectors"
+
+        NETPOL_EXT_NOTKUBE_COUNT=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.network.allow[] | select(.remoteGenerated != "IntraNamespace") | select(.remoteGenerated != "KubeAPI") | length' | wc -l)
+        echo "ℹ️ Non-IntraNamespace, non-KubeAPI network policies: $NETPOL_EXT_NOTKUBE_COUNT"
+
+        NETPOL_EXT_NOTKUBE_PORT_COUNT=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.network.allow[] | select(.remoteGenerated != "IntraNamespace") | select(.remoteGenerated != "KubeAPI") | select(.ports != null or .port != null) | length' | wc -l)
+        echo "ℹ️ Non-IntraNamespace, non-KubeAPI network policies with ports: $NETPOL_EXT_NOTKUBE_PORT_COUNT"
+
+        if [ "$NETPOL_EXT_NOTKUBE_COUNT" -ne "$NETPOL_EXT_NOTKUBE_PORT_COUNT" ]; then
+          echo "❌ Not all applicable network policies are using ports"
+          exit 1
+        fi
+        echo "✅ All applicable network policies are using ports"
+
```

Comment:
We should probably alert on `remoteGenerated: Anywhere` as a warning

Reply:
Added!